ee/tabby-webserver/src/repositories/resolve.rs

```diff
@@ -17,7 +24,47 @@ use tower_http::services::ServeDir;
 use crate::repositories::ResolveState;
 
 lazy_static! {
-    static ref META: HashMap<RepositoryKey, RepositoryMeta> = load_meta();
+    static ref META: RepositoryCache = RepositoryCache::new_initialized();
+}
+
+pub fn reload_repository_cache() {
+    META.reload()
+}
+
+pub fn start_reload_job() {
+    std::thread::spawn(|| {
```

Comment:
instead of starting a new thread, could you could a functino like `initialize` to wrap the logic in tokio::spawn? just like https://github.com/TabbyML/tabby/blob/673768e4cb2fab6dfa6c987176aa6c9fdc02da80/ee/tabby-webserver/src/service/cron/mod.rs#L19

Then you could invoke initialize in `ServerContext` (I'll also recommend to refactor the `META` out, putting `RepositoryCache` as part of member variable of `ServerContext`

Reply:
Working on that, this isn't ready for review yet